regression/cloud_providers.yaml

```diff
@@ -0,0 +1,27 @@
+name: streaming-regression-test-cloud-providers
+compute:
+  cluster: r1z1
+gpu_num: 8
+command: |-
+  pip uninstall -y mosaicml-streaming
+  cd streaming
+  pip install -e '.[dev]'
+  composer -n 1 regression/synthetic_dataset.py --create --cloud gs
+  composer -n 2 regression/iterate_data.py --cloud gs
+  composer -n 1 regression/synthetic_dataset.py --delete --cloud gs
+  composer -n 1 regression/synthetic_dataset.py --create --cloud s3
+  composer -n 4 regression/iterate_data.py --cloud s3
```

Comment:
What are we validating as part of this run ? I am thinking if we can validate the total number of shard files between remote and local directory or validate the total disk size used by the dataset between remote and local directory ?